airbyte-integrations/connectors/source-github/source_github/availability_strategies.py

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from typing import Dict, Optional
+
+import requests
+from airbyte_cdk.sources import Source
+from airbyte_cdk.sources.streams.core import Stream
+from airbyte_cdk.sources.streams.http.availability_strategy import HttpAvailabilityStrategy
+from airbyte_cdk.sources.utils.stream_helpers import StreamHelper
+from requests import HTTPError
+
+
+class OrganizationBasedAvailabilityStrategy(HttpAvailabilityStrategy):
+    """
+    Availability Strategy for organization-based streams.
+    """
+
+    def reasons_for_unavailable_status_codes(
+        self, stream: Stream, logger: logging.Logger, source: Optional["Source"], error: HTTPError
+    ) -> Dict[int, str]:
+        stream_slice = StreamHelper().get_stream_slice(stream)
+        organisation = stream_slice.get("organization", "")
+        response_error_msg = str(error.response.json().get("message"))
+
+        reasons_for_codes = {
+            requests.codes.NOT_FOUND: f"`{stream.__class__.__name__}` stream isn't available for organization `{stream_slice['organization']}`.",
+            # When `403` for the stream, that has no access to the organization's teams, based on OAuth Apps Restrictions:
+            # https://docs.github.com/en/organizations/restricting-access-to-your-organizations-data/enabling-oauth-app-access-restrictions-for-your-organization
+            requests.codes.FORBIDDEN: f"`{stream.name}` stream isn't available for organization `{organisation}`. Full error message: {response_error_msg}",
+        }
+        return reasons_for_codes
+
+
+class RepositoryBasedAvailabilityStrategy(HttpAvailabilityStrategy):
+    """
+    Availability Strategy for repository-based streams.
+    """
+
+    def reasons_for_unavailable_status_codes(
+        self, stream: Stream, logger: logging.Logger, source: Optional["Source"], error: HTTPError
+    ) -> Dict[int, str]:
+        stream_slice = StreamHelper().get_stream_slice(stream)
+        repository = stream_slice.get("repository", "")
```

Comment:
I new implementation we can be more strict now

```
stream_slice["repository"]
```

instead of `stream_slice.get("repository", "")`